bluemira/equilibria/coils/_grouping.py

```diff
@@ -0,0 +1,1088 @@
+        self.__setter("dx", values)
+
+    @dz.setter
+    def dz(self, values: Union[float, Iterable[float]]):
+        """Set coil dz sizes"""
+        self.__setter("dz", values)
+
+    @current.setter
+    def current(self, values: Union[float, Iterable[float]]):
+        """Set coil currents"""
+        self.__setter("current", values)
+
+    @j_max.setter
+    def j_max(self, values: Union[float, Iterable[float]]):
+        """Set coil max current densities"""
+        self.__setter("j_max", values)
+
+    @b_max.setter
+    def b_max(self, values: Union[float, Iterable[float]]):
+        """Set coil max fields"""
+        self.__setter("b_max", values)
+
+    @discretisation.setter
+    def discretisation(self, values: Union[float, Iterable[float]]):
+        """Set coil discretisations"""
+        self.__setter("discretisation", values)
+        self._pad_discretisation(self.__list_getter("_quad_x"))
+
+    @n_turns.setter
+    def n_turns(self, values: Union[float, Iterable[float]]):
+        """Set coil number of turns"""
+        self.__setter("n_turns", values)
+
+
+class Circuit(CoilGroup):
+    """
+    A CoilGroup with the same currents
+
+    Parameters
+    ----------
+    coils: Union[Coil, CoilGroup[Coil]]
+        coils in circuit
+    current: Optional[float]
+        The current value, if not provided the first coil current is used
+    """
+
+    def __init__(
+        self, *coils: Union[Coil, CoilGroup[Coil]], current: Optional[float] = None
+    ):
+        super().__init__(*coils)
+        self.current = self._get_current() if current is None else current
+
+    def _get_current(self):
+        current = self._coils[0].current
+        if isinstance(current, Iterable):
+            current = current[0]
+        return current
+
+    def add_coil(self, *coils: Union[Coil, CoilGroup[Coil]]):
+        """
+        Add coil to circuit forcing the same current
+        """
+        super().add_coil(coils)
+        self.current = self._get_current()
+
+    @CoilGroup.current.setter
+    def current(self, values: Union[float, Iterable[float]]):
+        """
+        Set current for circuit
+        """
+        if isinstance(values, Iterable):
+            # Force the same value of current for all coils
+            values = values[0]
+        self._CoilGroup__setter("current", values)
+
+
+class SymmetricCircuit(Circuit):
+    """
+    A Circuit with positional symmetry
+
+    Parameters
+    ----------
+    coils: Union[Coil, CoilGroup[Coil]]
+        2 coil or coil group objects to symmetrise in a circuit
+    symmetry_line: Union[Tuple, np.ndarray]
+        Symmetry line, by default the line z=0
+
+    Notes
+    -----
+    Although two groups can be defined any movement of the coils is
+    achieved by offsets to the mean position of the coilgroups
+    """
+
+    def __init__(
+        self,
+        *coils: Union[Coil, CoilGroup[Coil]],
+        symmetry_line: Union[Tuple, np.ndarray] = ((0, 0), (1, 0)),
```

Comment:
At present, symmetry will only work about the z=0 line. I would say don't expose this as an option until we bother to address https://github.com/Fusion-Power-Plant-Framework/bluemira/issues/210 (low priority, seeing as it's not that difficult to shift the grid).

Reply:
Would it not be possible to do eg x=5 as the symmetry line (or anything perpendicular to z, symmetric about something similar to the plasma centre)?

Reply:
Removed